racer.el

```diff
@@ -112,6 +114,242 @@
+          (push (propertize text 'face 'racer-help-heading-face)
+                sections)))
+       ;; For cross references, e.g. [`str`]: ../../std/primitive.str.html
+       ;; we simply skip over them.
+       ((and (not in-code) (string-match-p (rx bol "[`" (+? anything) "`]: ") line)))
+       ;; Skip repeated blank lines (caused by skipping cross references).
+       ((and (equal line "") (equal (-first-item current-section-lines) "")))
+       ;; Otherwise, just keep appending the line to the current
+       ;; section.
+       (t
+        (push line current-section-lines))))
+    (funcall finish-text-section)
+    (s-join "\n" (nreverse sections))))
+
+(defun racer--find-file (path line column)
+  "Open PATH and move point to LINE and COLUMN."
+  (find-file path)
+  (goto-char (point-min))
+  (forward-line (1- line))
+  (forward-char column))
+
+(defun racer--button-go-to-src (button)
+  (racer--find-file
+   (button-get button 'path)
+   (button-get button 'line)
+   (button-get button 'column)))
+
+(define-button-type 'racer-src-button
+  'action 'racer--button-go-to-src
+  'follow-link t
+  'help-echo "Go to definition")
+
+(defun racer--url-button (text url)
+  "Return a button that opens a browser at URL."
+  (with-temp-buffer
+    (insert-text-button
+     text
+     :type 'help-url
+     'help-args (list url))
+    (buffer-string)))
+
+(defun racer--src-button (path line column)
+  "Return a button that navigates to PATH at LINE number and
+COLUMN number."
+  ;; Convert "/foo/bar/baz/foo.rs" to "baz/foo.rs"
+  (let* ((filename (f-filename path))
+         (parent-dir (f-filename (f-parent path)))
+         (short-path (f-join parent-dir filename)))
+    (with-temp-buffer
+      (insert-text-button
+       short-path
+       :type 'racer-src-button
+       'path path
+       'line line
+       'column column)
+      (buffer-string))))
+
+(defun racer-describe ()
+  "Show a *Racer Help* buffer for the function or type at point."
+  (interactive)
+  (let ((description
+         (save-excursion
+           (skip-syntax-forward "w_")
+           (racer--describe-at-point))))
+    (unless description
+      (user-error "No function or type found at point"))
+    (let* ((name (plist-get description :name))
+           (raw-docstring (plist-get description :docstring))
+           (docstring (if raw-docstring
+                          (racer--propertize-docstring raw-docstring)
+                        "Not documented.")))
+      (switch-to-buffer
+       (racer--help-buf
+        name
+        (format
+         "%s is a %s defined in %s.\n\n%s\n\n%s"
+         name
+         (downcase (plist-get description :kind))
+         (racer--src-button
+          (plist-get description :path)
+          (plist-get description :line)
+          (plist-get description :column))
+         (concat "    " (racer--syntax-highlight (plist-get description :signature)))
+         docstring))))))
+
+(define-derived-mode racer-help-mode fundamental-mode
```

Comment:
Interesting idea! My first prototype tried to put the docstring in `*Help*` buffers, but it's really awkward to work with when you have propertized text.

Most of the help-mode bindings won't work (`help-go-back`, `help-go-forward`, `help-follow-symbol`, `help-mode-revert-buffer`). However, if we use `button-buffer-map` and `special-mode-map`, we get all the TAB and scrolling behaviour of help-mode :)